LCM/scripts/RemoveModule.py

```diff
@@ -4,124 +4,187 @@
 import subprocess
 import shutil
 import platform
+import imp
+
+from os.path import dirname, join, realpath
+
+pathToCurrentScript = realpath(__file__)
+pathToCommonScriptsFolder = dirname(pathToCurrentScript)
+
+helperLibPath = join(pathToCommonScriptsFolder, 'helperlib.py')
+helperlib = imp.load_source('helperlib', helperLibPath)
 
 def usage():
     print("Usage:")
     print("  RemoveModule.py NAME")
     sys.exit(1)
 
+def exitWithError(message, errorCode = 1):
+    errorMessage = "ERROR from RemoveModule.py: " + message
+    print(errorMessage)
+    sys.exit(errorCode)
 
-def RemoveMirroredDirectory(moduleDir, omiDir):
-    if len(moduleDir) < 15 or len(omiDir) < 10:
-        print("Error: The parameters to RemoveMirroredDirectory are too short.  This function can do dangerous things, so this is a sanity check.")
-        sys.exit(1)
-
-    for curFile in os.listdir(moduleDir):
-        if os.path.islink(moduleDir + "/" + curFile):
-            os.path.unlink(moduleDir + "/" + curFile)
-        elif os.path.isfile(moduleDir + "/" + curFile):
-            os.remove(moduleDir + "/" + curFile)
-        elif os.path.isdir(moduleDir + "/" + curFile):
-            RemoveMirroredDirectory(moduleDir + "/" + curFile, omiDir + "/" + curFile)
-            os.rmdir(moduleDir + "/" + curFile)
-            if len(os.listdir(omiDir + "/" + curFile)) == 0:
-                os.rmdir(omiDir + "/" + curFile)
-
-        if os.path.islink(omiDir + "/" + curFile):
-            os.path.unlink(omiDir + "/" + curFile)
-        elif os.path.isfile(omiDir + "/" + curFile):
-            os.remove(omiDir + "/" + curFile)
-
-if len(sys.argv) != 2:
-    usage()
-
-moduleName = sys.argv[1]
-omi_bindir = "<CONFIG_BINDIR>"
-omi_libdir = "<CONFIG_LIBDIR>"
-omi_sysconfdir = "<CONFIG_SYSCONFDIR>"
-baseModulePath = "<DSC_MODULES_PATH>"
-baseScriptPath = "<DSC_SCRIPT_PATH>"
-modulePath = baseModulePath + "/" + moduleName
-omiregister_dir = "/etc/opt/omi/conf/omiregister"
-
-if not os.path.isdir(modulePath):
-    print("Error: unable to find installed module " + moduleName + " at " + modulePath)
-    sys.exit(1)
+def printVerboseMessage(message):
+    verboseMessage = "VERBOSE from RemoveModule.py: " + message
+    print(verboseMessage)
 
-print("Removing module " + moduleName)
+def getPlatformArchitectureFolderName():
+    platformArchitecture = platform.architecture()
+    if len(platformArchitecture) != 2:
+        exitWithError("The python function platform.architecture() failed to return a valid tuple. Cannot detect if this system has an x64 or an x86 architecture.")
 
```

Comment:
Consider putting those in a separate script.  Those methods about formatting error message and have very little to do with Remove/Install module.

Reply:
The separate library script was already addressed above. I don't want to add more complexity to this PR then is already there. All the Python scripts need a huge rewrite and I would like to keep this PR focused on the SxS fix.

Reply:
Okay I see your reply to similar comment above. At least you could have written a class for this instead of declaring these functions in the global scope in every function which uses them.